File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;

//品牌接口
public interface BrandService {
	
	//查询所有品牌
	public List<TbBrand> findAll();
	
	//分页查询
	public PageResult findPage(int pageNum,int  pageSize);
	
	//新增
	public void add(TbBrand tbBrand);
	
	//根据id查询
	public TbBrand findOne(Long id);
	
	//修改对象
	public void update(TbBrand tbBrand);
	
	//删除对象
	public void delete(Long[] ids);
	
	//条件分页查询 方法重载
	public PageResult findPage(TbBrand tbBrand,int pageNum,int  pageSize);
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
 //定义品牌服务层
    app.service("brandService",function($http){
    	//抽取查询所有
    	this.findAll=function(){
    		return $http.get('../brand/findAll.do');
    	}
    	//抽取分页方法
    	this.findPage=function(page,rows){
    		return $http.get('../brand/findPage.do?page='+page+'&size='+rows);
    	}
    	//添加方法
    	this.add=function(entity){
    		return $http.post('../brand/save.do',entity);
    	}
    	//修改方法
    	this.update=function(entity){
            return $http.post('../brand/update.do',entity);
        }
    	//根据id查询
    	this.findOne=function(id){
    		return $http.get('../brand/findOne.do?id='+id);
    	}
    	//删除方法
    	this.delete=function(selectIds){
    		return $http.post('../brand/delete.do?ids='+selectIds);
    	}
    	//条件查询的方法
    	this.search=function(page,rows,searchEntity){
    		return $http.post('../brand/search.do?page='+page+'&size='+rows,searchEntity);
    	}
    });